src/Symfony/Component/Validator/Constraints/NullValidator.php

```diff
@@ -18,6 +18,8 @@
  * @author Bernhard Schussek <bschussek@gmail.com>
  *
  * @api
+ *
+ * @deprecated
```

Comment:
given you are replacing it with the IsNullValidator, the old class must be deprecated yes (and extend the new one to avoid duplicating the logic)

@nicolas-grekas the constraint validators are generally not instantiated by users anyway (especially when there is no dependency like here). The class kept for BC is only for hardcore users wanting to do weird things with the validator